atomicwrites/__init__.py

```diff
@@ -131,18 +139,19 @@ def __init__(self, path, mode='w', overwrite=False):
         if 'w' not in mode:
             raise ValueError('AtomicWriters can only be written to.')
 
-        self._path = path
-        self._mode = mode
-        self._overwrite = overwrite
+        self._path = _path_to_unicode(path)  # type: Text
+        self._mode = mode       # type: Text
+        self._overwrite = overwrite  # type: bool
 
-    def open(self):
+    def open(self):  # type: ignore
         '''
         Open the temporary file.
         '''
         return self._open(self.get_fileobject)
 
     @contextlib.contextmanager
     def _open(self, get_fileobject):
+        # type: (Callable) -> Iterator[IO]
```

Comment:
I'm not exactly sure what's going on here, but I think maybe it type-checks the un-decorated function, which, when read as written, is a generator that yields an IO object (i.e. an open file). So this is the return type that mypy accepts.